File: app/lib/screens/tela_prestador.dart
import 'package:flutter/material.dart';

import '../models/avaliacao.dart';
import '../models/prestador.dart';
import '../theme/tema.dart';
import '../widgets/avatar_prestador.dart';
import '../widgets/estrelas.dart';
import 'tela_avaliar.dart';

/// Perfil completo do prestador: reputação, sobre, catálogo e avaliações.
///
/// É Stateful porque a lista de avaliações pode mudar quando o cliente
/// deixa uma nova avaliação.
class TelaPrestador extends StatefulWidget {
  final Prestador prestador;

  const TelaPrestador({super.key, required this.prestador});

  @override
  State<TelaPrestador> createState() => _TelaPrestadorState();
}

class _TelaPrestadorState extends State<TelaPrestador> {
  Prestador get p => widget.prestador;

  /// Abre a tela de avaliar e, se o cliente enviar, adiciona à lista.
  Future<void> _avaliar() async {
    final nova = await Navigator.of(context).push<Avaliacao>(
      MaterialPageRoute(
        builder: (_) => TelaAvaliar(nomePrestador: p.nome),
      ),
    );
    if (nova != null) {
      setState(() => p.avaliacoes.insert(0, nova));
      if (mounted) {
        ScaffoldMessenger.of(context).showSnackBar(
          const SnackBar(content: Text('Obrigado pela avaliação! ⭐')),
        );
      }
    }
  }

  void _contratar() {
    // Numa próxima etapa: abrir chat interno ou WhatsApp.
    ScaffoldMessenger.of(context).showSnackBar(
      SnackBar(content: Text('Solicitação enviada para ${p.nome} (em breve)')),
    );
  }

  @override
  Widget build(BuildContext context) {
    final semAvaliacao = p.totalAvaliacoes == 0;

    return Scaffold(
      appBar: AppBar(title: Text(p.nome)),
      body: ListView(
        padding: const EdgeInsets.only(bottom: 24),
        children: [
          // ----- Cabeçalho com foto, nome, nota -----
          Container(
            color: Colors.white,
            padding: const EdgeInsets.all(16),
            child: Column(
              children: [
                AvatarPrestador(prestador: p, raio: 44),
                const SizedBox(height: 12),
                Row(
                  mainAxisAlignment: MainAxisAlignment.center,
                  children: [
                    Text(
                      p.nome,
                      style: const TextStyle(
                          fontSize: 20, fontWeight: FontWeight.bold),
                    ),
                    if (p.verificado) ...[
                      const SizedBox(width: 6),
                      const Icon(Icons.verified, color: Tema.verde, size: 20),
                    ],
                  ],
                ),
                const SizedBox(height: 4),
                Text(
                  p.especialidade,
                  textAlign: TextAlign.center,
                  style: TextStyle(color: Colors.grey[700]),
                ),
                const SizedBox(height: 8),
                if (semAvaliacao)
                  Text('Sem avaliações ainda',
                      style: TextStyle(color: Colors.grey[600]))
                else
                  Row(
                    mainAxisAlignment: MainAxisAlignment.center,
                    children: [
                      Estrelas(nota: p.mediaEstrelas, tamanho: 22),
                      const SizedBox(width: 8),
                      Text(
                        '${p.mediaEstrelas.toStringAsFixed(1)} · '
                        '${p.totalAvaliacoes} avaliações',
                        style: const TextStyle(fontWeight: FontWeight.w600),
                      ),
                    ],
                  ),
                const SizedBox(height: 6),
                Row(
                  mainAxisAlignment: MainAxisAlignment.center,
                  children: [
                    Icon(Icons.location_on, size: 16, color: Colors.grey[500]),
                    const SizedBox(width: 2),
                    Text(p.cidade,
                        style: TextStyle(color: Colors.grey[600])),
                  ],
                ),
              ],
            ),
          ),

          // ----- Sobre -----
          _Secao(
            titulo: 'Sobre',
            child: Text(p.sobre, style: const TextStyle(height: 1.4)),
          ),

          // ----- Catálogo de trabalhos -----
          _Secao(
            titulo: 'Trabalhos realizados',
            child: p.catalogo.isEmpty
                ? Text('Este profissional ainda não adicionou fotos.',
                    style: TextStyle(color: Colors.grey[600]))
                : SizedBox(
                    height: 110,
                    child: ListView.separated(
                      scrollDirection: Axis.horizontal,
                      itemCount: p.catalogo.length,
                      separatorBuilder: (_, __) => const SizedBox(width: 8),
                      itemBuilder: (_, i) => ClipRRect(
                        borderRadius: BorderRadius.circular(12),
                        child: Image.network(
                          p.catalogo[i],
                          width: 140,
                          fit: BoxFit.cover,
                        ),
                      ),
                    ),
                  ),
          ),

          // ----- Avaliações -----
          _Secao(
            titulo: 'Avaliações',
            child: semAvaliacao
                ? Text('Seja o primeiro a avaliar!',
                    style: TextStyle(color: Colors.grey[600]))
                : Column(
                    children: p.avaliacoes
                        .map((a) => _AvaliacaoItem(avaliacao: a))
                        .toList(),
                  ),
          ),
        ],
      ),

      // ----- Botões de ação fixos embaixo -----
      bottomNavigationBar: SafeArea(
        child: Padding(
          padding: const EdgeInsets.all(12),
          child: Row(
            children: [
              Expanded(
                child: OutlinedButton.icon(
                  onPressed: _avaliar,
                  icon: const Icon(Icons.star_outline),
                  label: const Text('Avaliar'),
                  style: OutlinedButton.styleFrom(
                    minimumSize: const Size.fromHeight(52),
                    foregroundColor: Tema.verde,
                    side: const BorderSide(color: Tema.verde),
                    shape: RoundedRectangleBorder(
                      borderRadius: BorderRadius.circular(14),
                    ),
                  ),
                ),
              ),
              const SizedBox(width: 12),
              Expanded(
                child: FilledButton.icon(
                  onPressed: _contratar,
                  icon: const Icon(Icons.chat_bubble_outline),
                  label: const Text('Contratar'),
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }
}

/// Bloco de conteúdo com um título, separado por um cartão branco.
class _Secao extends StatelessWidget {
  final String titulo;
  final Widget child;

  const _Secao({required this.titulo, required this.child});

  @override
  Widget build(BuildContext context) {
    return Container(
      width: double.infinity,
      color: Colors.white,
      margin: const EdgeInsets.only(top: 10),
      padding: const EdgeInsets.all(16),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Text(titulo,
              style:
                  const TextStyle(fontSize: 16, fontWeight: FontWeight.bold)),
          const SizedBox(height: 10),
          child,
        ],
      ),
    );
  }
}

/// Uma avaliação exibida na lista (nome, estrelas, comentário).
class _AvaliacaoItem extends StatelessWidget {
  final Avaliacao avaliacao;

  const _AvaliacaoItem({required this.avaliacao});

  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: const EdgeInsets.only(bottom: 14),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Row(
            children: [
              Text(avaliacao.nomeCliente,
                  style: const TextStyle(fontWeight: FontWeight.w600)),
              const Spacer(),
              Estrelas(nota: avaliacao.estrelas.toDouble(), tamanho: 15),
            ],
          ),
          if (avaliacao.comentario.isNotEmpty) ...[
            const SizedBox(height: 4),
            Text(avaliacao.comentario,
                style: TextStyle(color: Colors.grey[800], height: 1.3)),
          ],
        ],
      ),
    );
  }
}

File: android-app/app/src/main/java/com/servicos/app/ui/screens/TelaPrestador.kt
@file:OptIn(androidx.compose.material3.ExperimentalMaterial3Api::class)

package com.servicos.app.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material.icons.filled.ChatBubbleOutline
import androidx.compose.material.icons.filled.LocationOn
import androidx.compose.material.icons.filled.StarOutline
import androidx.compose.material.icons.filled.Verified
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.OutlinedButton
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Text
import androidx.compose.material3.TopAppBar
import androidx.compose.material3.TopAppBarDefaults
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import com.servicos.app.data.DadosMock
import com.servicos.app.model.Avaliacao
import com.servicos.app.model.Prestador
import com.servicos.app.ui.components.AvatarPrestador
import com.servicos.app.ui.components.Estrelas
import com.servicos.app.ui.theme.Verde

/** Perfil completo do prestador: reputação, sobre, catálogo e avaliações. */
@Composable
fun TelaPrestador(
    prestadorId: String,
    onVoltar: () -> Unit,
    onAvaliar: () -> Unit,
) {
    val prestador = DadosMock.prestadorPorId(prestadorId) ?: return
    val semAvaliacao = prestador.totalAvaliacoes == 0

    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text(prestador.nome) },
                navigationIcon = {
                    IconButton(onClick = onVoltar) {
                        Icon(Icons.AutoMirrored.Filled.ArrowBack, contentDescription = "Voltar")
                    }
                },
                colors = TopAppBarDefaults.topAppBarColors(
                    containerColor = Verde,
                    titleContentColor = Color.White,
                    navigationIconContentColor = Color.White,
                ),
            )
        },
        bottomBar = { BarraAcoes(prestador = prestador, onAvaliar = onAvaliar) },
    ) { padding ->
        LazyColumn(modifier = Modifier.padding(padding)) {
            // ----- Cabeçalho -----
            item {
                Column(
                    modifier = Modifier
                        .fillMaxWidth()
                        .background(Color.White)
                        .padding(16.dp),
                    horizontalAlignment = Alignment.CenterHorizontally,
                ) {
                    AvatarPrestador(prestador = prestador, tamanho = 88.dp)
                    Spacer(Modifier.size(12.dp))
                    Row(verticalAlignment = Alignment.CenterVertically) {
                        Text(
                            text = prestador.nome,
                            style = MaterialTheme.typography.headlineSmall,
                            fontWeight = FontWeight.Bold,
                        )
                        if (prestador.verificado) {
                            Spacer(Modifier.width(6.dp))
                            Icon(Icons.Filled.Verified, contentDescription = "Verificado", tint = Verde)
                        }
                    }
                    Text(
                        text = prestador.especialidade,
                        style = MaterialTheme.typography.bodyMedium,
                        color = Color.Gray,
                        textAlign = TextAlign.Center,
                    )
                    Spacer(Modifier.size(8.dp))
                    if (semAvaliacao) {
                        Text("Sem avaliações ainda", color = Color.Gray)
                    } else {
                        Row(verticalAlignment = Alignment.CenterVertically) {
                            Estrelas(nota = prestador.mediaEstrelas, tamanho = 22.dp)
                            Spacer(Modifier.width(8.dp))
                            Text(
                                text = "%.1f · %d avaliações".format(
                                    prestador.mediaEstrelas,
                                    prestador.totalAvaliacoes,
                                ),
                                fontWeight = FontWeight.SemiBold,
                            )
                        }
                    }
                    Spacer(Modifier.size(6.dp))
                    Row(verticalAlignment = Alignment.CenterVertically) {
                        Icon(Icons.Filled.LocationOn, contentDescription = null, tint = Color.Gray, modifier = Modifier.size(16.dp))
                        Spacer(Modifier.width(2.dp))
                        Text(prestador.cidade, color = Color.Gray)
                    }
                }
            }

            // ----- Sobre -----
            item {
                Secao(titulo = "Sobre") {
                    Text(prestador.sobre, style = MaterialTheme.typography.bodyMedium)
                }
            }

            // ----- Catálogo -----
            item {
                Secao(titulo = "Trabalhos realizados") {
                    Text(
                        text = "Este profissional ainda não adicionou fotos.",
                        color = Color.Gray,
                        style = MaterialTheme.typography.bodyMedium,
                    )
                }
            }

            // ----- Avaliações -----
            item {
                Secao(titulo = "Avaliações") {
                    if (semAvaliacao) {
                        Text("Seja o primeiro a avaliar!", color = Color.Gray)
                    } else {
                        Column {
                            prestador.avaliacoes.forEach { AvaliacaoItem(it) }
                        }
                    }
                }
            }
        }
    }
}

/** Barra fixa embaixo com os botões Avaliar e Contratar. */
@Composable
private fun BarraAcoes(prestador: Prestador, onAvaliar: () -> Unit) {
    Row(
        modifier = Modifier
            .background(Color.White)
            .fillMaxWidth()
            .padding(12.dp),
        horizontalArrangement = Arrangement.spacedBy(12.dp),
    ) {
        OutlinedButton(
            onClick = onAvaliar,
            modifier = Modifier.weight(1f),
        ) {
            Icon(Icons.Filled.StarOutline, contentDescription = null)
            Spacer(Modifier.width(6.dp))
            Text("Avaliar")
        }
        Button(
            onClick = { /* Em breve: chat interno ou WhatsApp */ },
            colors = ButtonDefaults.buttonColors(containerColor = Verde),
            modifier = Modifier.weight(1f),
        ) {
            Icon(Icons.Filled.ChatBubbleOutline, contentDescription = null)
            Spacer(Modifier.width(6.dp))
            Text("Contratar")
        }
    }
}

/** Bloco de conteúdo com título, sobre um fundo branco. */
@Composable
private fun Secao(titulo: String, conteudo: @Composable () -> Unit) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .padding(top = 10.dp)
            .background(Color.White)
            .padding(16.dp),
    ) {
        Text(titulo, style = MaterialTheme.typography.titleMedium, fontWeight = FontWeight.Bold)
        Spacer(Modifier.size(10.dp))
        conteudo()
    }
}

/** Uma avaliação exibida na lista (nome, estrelas, comentário). */
@Composable
private fun AvaliacaoItem(avaliacao: Avaliacao) {
    Column(modifier = Modifier.padding(bottom = 14.dp)) {
        Row(verticalAlignment = Alignment.CenterVertically) {
            Text(avaliacao.nomeCliente, fontWeight = FontWeight.SemiBold)
            Spacer(Modifier.weight(1f))
            Estrelas(nota = avaliacao.estrelas.toDouble(), tamanho = 15.dp)
        }
        if (avaliacao.comentario.isNotEmpty()) {
            Spacer(Modifier.size(4.dp))
            Text(avaliacao.comentario, color = Color.DarkGray, style = MaterialTheme.typography.bodyMedium)
        }
    }
}
